src/database/mod.rs

```diff
@@ -393,21 +367,41 @@ impl Database {
 pub enum Error {
     #[error("database error: {0}")]
     InternalError(#[from] sqlx::Error),
+
+    #[error("Tried to mine missing root {root:?}")]
+    MissingRoot { root: Hash },
 }
 
 #[cfg(test)]
 mod test {
     use std::time::Duration;
 
+    use anyhow::Context;
     use chrono::Utc;
+    use postgres_docker_utils::DockerContainerGuard;
     use reqwest::Url;
     use semaphore::Field;
 
     use super::{Database, Options};
     use crate::identity_tree::Status;
 
-    #[tokio::test]
-    async fn test_root_invalidation() -> anyhow::Result<()> {
+    macro_rules! assert_same_time {
+        ($a:expr, $b:expr, $diff:expr) => {
+            assert!(
+                $a - $b < $diff,
```

Comment:
`chrono` allows negative durations, so you probably also want to check `$b -$a < $diff`. In fact, I think most uses of this macro in these tests are vacuous, in that `$a` is usually before `$b`.